mdata/aggmetric_test.go

```diff
@@ -302,6 +308,156 @@ func TestAggMetricDropFirstChunk(t *testing.T) {
 	}
 }
 
+func TestAggMetricIngestFrom(t *testing.T) {
+	cluster.Init("default", "test", time.Now(), "http", 6060)
+	cluster.Manager.SetPrimary(true)
+	mockstore.Reset()
+	chunkSpan := uint32(10)
+	numChunks := uint32(5)
+	ingestFrom := int64(25)
+	ret := []conf.Retention{conf.NewRetentionMT(1, 1, chunkSpan, numChunks, 0)}
+	m := NewAggMetric(mockstore, &cache.MockCache{}, test.GetAMKey(42), ret, 0, 1, nil, false, ingestFrom)
+	m.Add(10, 10)
+	m.Add(11, 11)
+	m.Add(12, 12)
+	m.Add(20, 20)
+	m.Add(21, 21)
+	m.Add(22, 22)
+	m.Add(30, 30)
+	m.Add(31, 31)
+	m.Add(32, 32)
+	// the chunk that point belongs to is not returned by the search because the chunk has
+	// not been closed yet which will happen if a point belonging to the following chunk is added
+	m.Add(40, 40)
+	itgens, err := mockstore.Search(test.NewContext(), test.GetAMKey(42), 0, 0, 1000)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if len(itgens) != 1 || itgens[0].T0 != 30 {
+		t.Fatalf("expected 1 itgens for chunk 30 since the chunks before 25 should be dropped. Got %v", itgens)
+	}
+}
+
+func itersToPoints(iters []tsz.Iter) []schema.Point {
+	points := make([]schema.Point, 0)
```

Comment:
https://dave.cheney.net/2013/01/19/what-is-the-zero-value-and-why-is-it-useful

Reply:
Replaced with nil in commit ad7d980